### PR TITLE
Revise Edition introduction

### PR DESCRIPTION
**Pull request by aturon**
This PR refactors the intro text, mostly to change emphasis by talking about bottom-line impact first.

### CODE REVIEW

**Comment by aturon** [comment]
One thing I noticed: in this section, we talk about the idioms feature, but in the transition guide, we don't. Was that intended?

File: src/editions/index.md [hunk 1]
```diff
@@ -1,37 +1,36 @@
...
+
+This serves different purposes for different people:
+
+- For active Rust users, it brings together incremental changes into an
+  easy-to-understand package.
+
+- For non-users, it signals that some major advancements have landed, which
+  might make Rust worth another look.
+
+- For those working *on* Rust, it provides a rallying point for the project as a
```
Comment (Centril) [inline src/editions/index.md]: Perhaps reword `working on Rust` as `developing Rust itself` ?

File: src/editions/index.md [hunk 2]
```diff
@@ -1,37 +1,36 @@
...
-Editions are allowed to introduce minor kinds of incompatibilities, such as
-turning a warning into a hard error, or adding new keywords. At the same
-time, it's very important that we don't split the Rust ecosystem. Therefore,
-each project can list which edition that it's using, and the Rust compiler
-can link crates of any editions together. Therefore, if you're using Rust 2015,
-and one of your dependencies uses Rust 2018, it all works just fine. The
-opposite situation works as well.
+When a new edition becomes available in the compiler, crates must explicitly opt
+in to it to take full advantage. This opt in makes it possible to make
+incompatible changes, like adding a new keyword (which might conflict with
```
Comment (Centril) [inline src/editions/index.md]: Style nit: comma instead of parens ;)

File: src/editions/index.md [hunk 3]
```diff
@@ -40,19 +39,23 @@ Rust; Rust 2018 will ship later this year. To give Rust 2018 a try, you can
...
+
+```toml
+cargo-features = ["edition"]
+
+[package]
+edition = '2018'
+```
 
+Like all `#![feature(..)]` flags, this will only work on nightly Rust.
 As new nightly compilers are released, more functionality will be enabled,
```
Comment (Centril) [inline src/editions/index.md]: Not part of your changes, but... I suggest rewording this as:

> When new nightly compilers are released, more functionality is enabled.
> As a result, you may experience new warnings, features, and other things.

File: src/editions/index.md [hunk 4]
```diff
@@ -1,37 +1,36 @@
...
 
-Editions are allowed to introduce minor kinds of incompatibilities, such as
-turning a warning into a hard error, or adding new keywords. At the same
-time, it's very important that we don't split the Rust ecosystem. Therefore,
-each project can list which edition that it's using, and the Rust compiler
-can link crates of any editions together. Therefore, if you're using Rust 2015,
-and one of your dependencies uses Rust 2018, it all works just fine. The
-opposite situation works as well.
+When a new edition becomes available in the compiler, crates must explicitly opt
+in to it to take full advantage. This opt in makes it possible to make
```
Comment (otavio) [inline src/editions/index.md]: ... This opt-in, make it possible to make ...

Comment (Mark-Simulacrum) [reply src/editions/index.md]: "Opting into the edition permits us to make backwards incompatible changes"

**Comment by aturon** [comment]
Comments addressed! I'm going to be in meetings the rest of the day, so I suggest if there are further changes those be handled in follow-up PRs.
